Lib/traceback.py

```diff
@@ -498,8 +499,15 @@ def __init__(self, exc_type, exc_value, exc_traceback, *, limit=None,
                 _seen=_seen)
         else:
             cause = None
-        if (exc_value and exc_value.__context__ is not None
-            and id(exc_value.__context__) not in _seen):
+
+        self.__suppress_context__ = \
+            exc_value.__suppress_context__ if exc_value else False
```

Comment:
```suggestion
            exc_value.__suppress_context__ if exc_value is not None else False
```
(Some joker could add a `__bool__` method to their exception that sometimes  makes it falsey.)